src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncReceiptsRequestHandler.scala

```diff
@@ -0,0 +1,46 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor.{Scheduler, Props, ActorRef}
+import akka.util.ByteString
+import io.iohk.ethereum.db.storage.ReceiptStorage
+import io.iohk.ethereum.network.p2p.messages.PV63.{Receipts, GetReceipts}
+
+class FastSyncReceiptsRequestHandler(
+    peer: ActorRef,
+    requestedHashes: Seq[ByteString],
+    receiptStorage: ReceiptStorage)(implicit scheduler: Scheduler)
+  extends FastSyncRequestHandler[GetReceipts, Receipts](peer) {
+
+  override val requestMsg = GetReceipts(requestedHashes)
+  override val responseMsgCode = Receipts.code
+
+  override def handleResponseMsg(receipts: Receipts): Unit = {
+    (requestedHashes zip receipts.receiptsForBlocks).foreach { case (hash, receiptsForBlock) =>
+      receiptStorage.put(hash, receiptsForBlock)
+    }
+
+    if (receipts.receiptsForBlocks.isEmpty) {
```

Comment:
we are not removing them. it's the response from peer, we save all received receipts above. here is the check if the response was empty (then we blacklist)